lib/services/local/users.go

```diff
@@ -578,14 +579,30 @@ func (s *IdentityService) UpsertWebauthnLocalAuth(ctx context.Context, user stri
 		return trace.Wrap(err)
 	}
 
-	value, err := json.Marshal(wla)
+	wlaJSON, err := json.Marshal(wla)
 	if err != nil {
-		return trace.Wrap(err)
+		return trace.Wrap(err, "marshal webauthn local auth")
 	}
-	_, err = s.Put(ctx, backend.Item{
+	if _, err = s.Put(ctx, backend.Item{
 		Key:   webauthnLocalAuthKey(user),
-		Value: value,
+		Value: wlaJSON,
+	}); err != nil {
+		return trace.WrapWithMessage(err, "writing webauthn local auth")
+	}
+
+	userJSON, err := json.Marshal(&wantypes.User{
+		TeleportUser: user,
 	})
+	if err != nil {
+		return trace.WrapWithMessage(err, "marshal webauthn user")
+	}
+	if _, err = s.Put(ctx, backend.Item{
```

Comment:
With this change, it seems like we now have to keep values under webauthnLocalAuthKey and webauthnUserKey in sync. I.e. what happens if the Put call above succeeds but this one fails? Will the database be in inconsistent state? Do we need to rollback the above Put in case of failure here?

Reply:
We have to deal with index inconsistencies due to legacy entries, so it's nothing new. To avoid funky behavior, if this fails we have to fail registration of passwordless devices (which does happen), so we should be alright.

I'll leave a comment, but let me know if you prefer more decisive action.

Reply:
I actually went back on this one and included an attempt to "rollback" the first write. PTAL.